src/warc2zim/content_rewriting/generic.py

```diff
@@ -122,6 +122,64 @@ def rewrite(
         return ("", self.content)
 
     def get_rewrite_mode(self, record, mimetype):
+        """Get current record rewrite mode
+
+        The rewrite mode is used to decide which kind of resource we have (html, css,
+        js, ...) and this is used to decide how it should be parsed and rewritten.
+        """
+        mimetype_rewrite_mode = self.get_mimetype_rewrite_mode(record, mimetype)
+
+        resourcetype = record.rec_headers["WARC-Resource-Type"]
```

Comment:
I dont know the type of rec_headers. Is it safe to access this way? ie. a `dict` would raise `KeyError` but I suppose this is not a `dict`.

Reply:
Indeed, it is a [StatusAndHeaders](https://github.com/webrecorder/warcio/blob/de769a306d945d2beacb11727144f10932046e0e/warcio/statusandheaders.py#L15) object, and it is safe to access this way (it is even not case-sensitive). I'm 100% sure because all test WARC except the one I added in this PR do not have the `WARC-Resource-Type` header since they have been built before this header was introduced. Should this be a concern, tests would have failed.